Kitodo/src/main/java/org/kitodo/production/handler/RangeStreamedContentHandler.java

```diff
@@ -0,0 +1,297 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production.handler;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.el.ValueExpression;
+import javax.faces.context.ExternalContext;
+import javax.faces.context.FacesContext;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.primefaces.application.resource.BaseDynamicContentHandler;
+import org.primefaces.model.StreamedContent;
+import org.primefaces.util.Constants;
+
+public class RangeStreamedContentHandler extends BaseDynamicContentHandler {
+
+    private static final int DEFAULT_BUFFER_SIZE = 20480; // ..bytes = 20KB.
+
+    private static final String MULTIPART_BOUNDARY = "MULTIPART_BYTERANGES";
+
+    private static final String CLIENT_ABORT_EXCEPTION_CANONICAL_NAME = "org.apache.catalina.connector.ClientAbortException";
+
+    private static final Logger LOGGER = Logger.getLogger(RangeStreamedContentHandler.class.getName());
+
+    @Override
+    public void handle(FacesContext context) throws IOException {
+        Map<String, String> params = context.getExternalContext().getRequestParameterMap();
+        String library = params.get("ln");
+        String resourceKey = params.get(Constants.DYNAMIC_CONTENT_PARAM);
+
+        if (resourceKey != null && library != null && library.equals(Constants.LIBRARY)) {
```

Comment:
Why not use `Objects.nonNull(...)` as replacement for or `... != null` and / or `Objects.isNull(...)` as replacement for ` ... == null` here and in all other places?